modules/accredited_representative_portal/app/controllers/accredited_representative_portal/v0/form21a_controller.rb

```diff
@@ -22,14 +31,33 @@ def submit
 
       attr_reader :parsed_request_body
 
+      def schema
+        # NOTE: This doesn't reject any extra attributes not found in the schema. If
+        # we want that the schema needs to have { "additionalProperties" => false }
+        VetsJsonSchema::SCHEMAS[FORM_ID.upcase]
+      end
+
       # Parses the raw request body as JSON and assigns it to an instance variable.
       # Renders a bad request response if the JSON is invalid.
       def parse_request_body
         @parsed_request_body = JSON.parse(request.raw_post)
       rescue JSON::ParserError
-        Rails.logger.error(
-          "Form21aController: Invalid JSON in request body for user with user_uuid=#{@current_user&.uuid}"
-        )
+        handle_json_error
+      end
+
+      def validate_schema
+        errors = JSON::Validator.fully_validate(schema, parsed_request_body)
+        raise SchemaValidationError, errors if errors.any?
+      rescue SchemaValidationError => e
+        handle_json_error(e.errors.join(', ').squeeze(' '))
+      end
```

Comment:
Are we mediating with this new `SchemaValidationError` class rather than inlining the logic to set us up to eventually have some centralized error handling with `rescue_from`'s? I think that could be a good premature structuring but that it is marginal. Just curious on your thinking there.

Reply:
I actually had a bigger refactor of that controller at first that I scrapped but yeah I was centralizing the error handling. I scrapped because I opted for the smallest amount of changes necessary for this card and maybe just do a refactor in the future and ended up keeping that Error class. But you're absolutely right that I could've just inlined that part. 😅

Reply:
Cool, keeping as is is fine.